### PR TITLE
Fixes cleanbot renaming, buffs emagged cleanbots w/ knives

### LINKED ISSUE #47810: Cleanbot dosent keep his assigned name after stabbing someone
1. Name the Assembly
2. assign knife
3. it dosent Keep ist name

Reporting client version: 512.1485

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
Fixes #47810 where cleanbots would fallback to the name "cleanbot" when deputized
Reduces time to attach knife from 4 seconds to 2.5 seconds (4 seconds was juuuuuuust long enough to let it move while you were attaching)
Emagged cleanbots will do full damage when they stab someone rather than half damage

## Why It's Good For The Game
Makes knifebots more robust

## Changelog
:cl: Ryll/Shaps
balance: Cleanbots with knives will do full damage rather than half if emagged
fix: Cleanbots will no longer lose their name when given knives
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/cleanbot.dm [hunk 1]
```diff
@@ -56,8 +57,8 @@
 		user.transferItemToLoc(W, src)
 		weapon = W
 		weapon_orig_force = weapon.force
-		weapon.force = weapon.force / 2
-		icon_state = "cleanbot[on]"
+		if(!emagged)
+			weapon.force = weapon.force / 2
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/bot/cleanbot.dm]: what if you attach a knife to an already emagged one

Comment (Shapsy) [reply code/modules/mob/living/simple_animal/bot/cleanbot.dm]: then it stays at full force, which is the intended behavior. The other possibility (emagging an already armed bot) is taken care of on line 170

File: code/modules/mob/living/simple_animal/bot/cleanbot.dm [hunk 2]
```diff
@@ -157,14 +159,16 @@
...
-		if(do_after(user, 40, target = src))
+		to_chat(user, "<span class='notice'>You start attaching \the [W] to \the [src]...</span>")
+		if(do_after(user, 25, target = src))
 			deputize(W, user)
 	else
 		return ..()
 
 /mob/living/simple_animal/bot/cleanbot/emag_act(mob/user)
 	..()
+	if(weapon)
```
Comment (GunHog) [inline code/modules/mob/living/simple_animal/bot/cleanbot.dm]: Note that if a silicon hacks an armed cleanbot, the weapon force will not be affected, as this does not called `emag_act()` and can instead be found in the parent `Topic()` proc. However, if someone deputizes the cleanbot while in the hacked state, the weapon will have full force due to the emag check above. Sorry to burden you with this.

By the way, the `emagged` var has two levels. 1 means the ID lock has been removed, 2 means the bot has been fully emagged and will begin hostile actions. Although, you might want to have a full power stabby cleanbot puttering about without it making obvious signs of being emagged.

Comment (Shapsy) [reply code/modules/mob/living/simple_animal/bot/cleanbot.dm]: blah good catch, I'm on it

**Pull request merged**